nknd.go

```diff
@@ -396,22 +396,41 @@ func GetID(seeds []string, publickey []byte) ([]byte, error) {
 		return id, nil
 	}
 
-	log.Infof("get ID from localnode faild, no ID in ledger: %v", err)
+	if err != nil {
+		log.Errorf("get ID from local ledger error: %v", err)
+	} else {
+		log.Infof("get no ID from local ledger")
+	}
 
 	rand.Shuffle(len(seeds), func(i int, j int) {
 		seeds[i], seeds[j] = seeds[j], seeds[i]
 	})
 
-	for _, seed := range seeds {
-		id, err := client.GetID(seed, publickey)
-		if err == nil && len(id) == config.NodeIDBytes {
-			return id, nil
+	n := uint32(len(seeds))
+	if n > config.Parameters.MaxGetIDSeeds {
+		n = config.Parameters.MaxGetIDSeeds
+	}
+
+	counter := make(map[string]uint32)
+	for i := uint32(0); i < n; i++ {
+		id, err := client.GetID(seeds[i], publickey)
+		if err == nil && id != nil {
+			counter[string(id)]++
+		} else {
+			counter[""]++
 		}
+	}
 
-		log.Warningf("get ID from %s met error: %v", seed, err)
+	for idStr, count := range counter {
+		if count > n/2 {
+			if idStr == "" {
+				return nil, nil
+			}
+			return []byte(idStr), nil
+		}
```

Comment:
Why GetID from multi-seed and select the majority response?
Node will JoinNet fail with a fake ID got from malicious seed. And will restart nknd to request next random seed from seedList.

Reply:
He will not restart nknd if at least one of his neighbor is malicious and do not reject connection. Seed node can return id arbitrarily so if seed is malicious, he can guarantee the node that is joining has at least one malicious neighbor and thus will not restart.

On the other hand, we may want to provide a long seed list. The previous logic will cause the node to query every seed when id is not generated yet.